### PR TITLE
[12.0][ADD] account_move_line_drilldown

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Installation
============

To allow the user to browse through the first two levels of the chart of account
codes, this module adds two dedicated columns to the journal item table to
store top level, and the top - 1 level of the account group hierarchy that the
journal item belongs to. The columns are populated automatically during the
installation of the module. The installation will take about 2 minutes for every
million move lines in your database.

Configuration
=============

In order to enable the pseudohierachy that this module exposes in its move
line pivot view, you need to configure the account group on all accounts in
your chart of accounts.

Usage
=====

Go to Invoicing -> Reporting -> Management -> Journal Item Drilldown to browse
journal items across the first two levels of the account group hierarchy plus
the account level.

By default, you will get a pivot view with the three levels expanded:

image

The three levels are: the top level of the account.group hierarchy,
the top - 1 level of the account.group hierarchy, and the account
itself. You can easily collapse a level by deselecting them in the `Group By`
dropdown in the search bar.

You can also browse the journal items by switching to the tree view and click
through the grouping levels in that view.

image

### CODE REVIEW

File: account_move_line_drilldown/models/account_move_line.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    account_root_code = fields.Char(
+        compute="_compute_account_codes", index=True, store=True)
+    account_sub_code = fields.Char(
+        compute="_compute_account_codes", index=True, store=True)
+
+    @api.depends("account_id", "account_id.code")
```
Comment (hbrunn) [inline account_move_line_drilldown/models/account_move_line.py]: why not just `"account_id.code"`?

File: account_move_line_drilldown/views/account_move_line_view.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <record id="account_move_line_drilldown_action" model="ir.actions.act_window">
+            <field name="name">Journal Item Drilldown</field>
+            <field name="res_model">account.move.line</field>
+            <field name="view_mode">tree,form</field>
+            <field name="context">{
+                "search_default_group_by_account_root_code": 1,
+                "search_default_group_by_account_sub_code": 1,
+                "search_default_account": 1,
+            }</field>
+    </record>
```
Comment (hbrunn) [inline account_move_line_drilldown/views/account_move_line_view.xml:24]: nice future extension would be a pivot view too based on these

**Comment by astirpe** [comment]
I tried on Runbot and works very well, for example in combination with other filters!
I think that an opinion from an accountant would be precious.

**Review by hbrunn — APPROVED** [review]

File: account_move_line_drilldown/models/account_move_line.py (hunk 1 above)
Comment (StefanRijnhart) [reply account_move_line_drilldown/models/account_move_line.py]: Thanks, fixed.

**Review by astirpe — APPROVED** [review]
@StefanRijnhart are you going to add something else in this PR? Or is it ready for review? I see the status is in draft...
IMO code review OK and tested on Runbot, so I think this is good enough for being merged

**Comment by StefanRijnhart** [comment]
@astirpe thank you for the review. I'm going to ask the accountants in the customer company to validate the functionality (c.q. usability) of the solution first.

File: account_move_line_drilldown/views/account_move_line_view.xml (hunk 2 above)
Comment (StefanRijnhart) [reply account_move_line_drilldown/views/account_move_line_view.xml:24]: Excellent, thanks! The pivot view is the best view to navigate this. Setting the pivot view as default pretty much solves the usability concerns I had about this solution. See the screenshot in the updated PR description.

**Comment by StefanRijnhart** [comment]
@OCA-git-bot please learn to respect the draft status of a PR.

**Comment by kittiu** [comment]
Nice feature @StefanRijnhart 
Still draft?

**Comment by StefanRijnhart** [comment]
@kittiu thanks! Indeed, the current iteration based on account.group was accepted by our business users, so I'm taking this PR out of draft state.

**Review by HaraldPanten — APPROVED** [review]
Functional review. I like it! 👍

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Pull request merged**